### PR TITLE
Fix issue with workflow

### PR DESCRIPTION
**Pull request by Jebay**

### CODE REVIEW

File: .github/workflows/gallery.yml [hunk 1]
```diff
@@ -8,21 +8,33 @@ on:
...
         uses: actions/checkout@v3
         with:
           persist-credentials: false
       - name: Build Gallery 🔧
         run: docker run -v "$(pwd):/work" thumbsupgallery/thumbsup /bin/sh -c "cd /work/ && thumbsup --config config.json"
+      - name: Package and upload artifact 📦
+        uses: actions/upload-pages-artifact@v1.0.7
+        with:
+          path: 'build_output'
+          name: gh-pages
```
Comment (Jebay) [inline .github/workflows/gallery.yml:31]: Fix the error `Error: No uploaded artifact was found! Please check if there are any errors at build step, or uploaded artifact name is correct.`

File: .github/workflows/gallery.yml [hunk 2]
```diff
@@ -8,21 +8,33 @@ on:
       - master
 permissions:
   contents: write
-
+  id-token: write
+  pages: write
```
Comment (Jebay) [inline .github/workflows/gallery.yml:12]: Fix the error `Failed to create deployment (status: 403). Ensure GITHUB_TOKEN has permission "pages: write".`

File: .github/workflows/gallery.yml [hunk 3]
```diff
@@ -8,21 +8,33 @@ on:
       - master
 permissions:
   contents: write
-
+  id-token: write
```
Comment (Jebay) [inline .github/workflows/gallery.yml:11]: Fix the error : `Ensure GITHUB_TOKEN has permission "id-token: write".
`

File: .github/workflows/gallery.yml [hunk 4]
```diff
@@ -8,21 +8,33 @@ on:
...
+  pages: write
 jobs:
   test:
     name: Publish Gallery to Github Pages
     if: github.repository != 'gautamkrishnar/github-pages-gallery'
     runs-on: ubuntu-latest
     steps:
+      - name: Setup GitHub Pages ⚙️
+        uses: actions/configure-pages@v3.0.4
```
Comment (Jebay) [inline .github/workflows/gallery.yml:20]: Ensure that GitHub Pages are enabled

File: .github/workflows/gallery.yml [hunk 5]
```diff
@@ -8,21 +8,33 @@ on:
...
+          name: gh-pages
       - name: Deploy to Github Pages 🚀
         uses: actions/deploy-pages@v1
         with:
           artifact_name: gh-pages
           token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Delete artifact 🗑️
+        uses: geekyeggo/delete-artifact@v2
+        with:
+          name: gh-page
```
Comment (Jebay) [inline .github/workflows/gallery.yml:40]: Delete uploaded artifact in order not to be blocked by the limitations of GitHub Storage.

At this time, there is no official GitHub Action for that

Comment (gautamkrishnar) [inline .github/workflows/gallery.yml:40]: please update to `name: gh-pages`

Comment (gautamkrishnar) [reply .github/workflows/gallery.yml:40]: will fix it myself later

**Comment by gautamkrishnar** [comment]
@Jebay thanks a lot for contributing. 🎉

**Pull request merged**